### DISCUSSION TITLE
package management for Monaco

### DISCUSSION THREAD

**Discussion by CruzanCaramele** [Dynatrace/dynatrace-configuration-as-code#517] [Ideas]
Make installation of monaco on different operating systems easier through the use of package managers.

Proposal:
Use package managers to install monaco based on operating system in use.

Windows:
Use **chocolatey**:
- **choco install monaco**
- **choco upgrade monaco**
- **choco uninstall monaco**
This will require building scripts and making monaco available  on **chocolatey.org**

macOS:
- **brew install monaco**
- **brew upgrade monaco**
- **brew uninstall monaco**
This will require building a formula and making it available on the **homebrew** repository

Debian based OS:
- **apt install monaco**
- **apt remove monaco**
This will require a respository in **https://github.com/dynatrace-oss** where we can provide this from.

The aim is to simplify installing, upgrading, and uninstallation of monaco. 
Most major tools can be installed using native package managers

@didiladi ( as discussed if approved, I will work on this)
@UnseenWizzard

**Comment by didiladi** [comment]
Cool, thanks @CruzanCaramele for entering this ticket. :muscle: 

Personally, I like the idea to make the install as seamless as possible to the user. There are some considerations/questions from my side:

* We should tackle signing of the binary first, before distributing it. We already have a ticket about the monaco exe being detected as a virus by some virus scanners. Iirc this happens for go binaries when the binary is not signed.  
* If going into that direction, we should automate the upload to the respective repository. Maybe there are existing GitHub actions we could use?
* Question: Is there anything in regards to branding/legal topics we need to consider? Are there any existing teams/binaries at Dynatrace which are already using this means of distribution?

@UnseenWizzard, @wlfghdr What is your take here?

**Reply by UnseenWizzard** [reply]
Great idea @CruzanCaramele!

And fully agree with what you mention in terms of caveats/considerations @didiladi

We could tackle the windows binary individually from the linux and mac though and provide packages sooner/independently. 

And +100 for looking for existing solutions/actions to building packages - I remember properly building deb and rpm packages (should consider more distros) as a bit of a pain

**Reply by CruzanCaramele** [reply]
👍 
> We could tackle the windows binary individually from the linux and mac though and provide packages sooner/independently.

For Windows after a build process, the executables could be signed using [microsoft signtool](https://docs.microsoft.com/en-us/windows/win32/seccrypto/signtool). These can then be used for building chocolatey packages.


> And +100 for looking for existing solutions/actions to building packages - I remember properly building deb and rpm packages (should consider more distros) as a bit of a pain

Yes certainly the process can be automated, example, using the Automatic Package Updater Module for chocolatey and using GH actions for homebrew and Linux packages
